libs/designsystem/src/lib/components/button/button.component.ts

```diff
@@ -102,7 +102,8 @@ export class ButtonComponent implements AfterContentInit {
     this.hasText = !!this.elementRef.nativeElement.textContent;
     if (this.iconElementRef !== undefined && this.hasText) {
       this._isIconLeft =
-        this.elementRef.nativeElement.firstChild === this.iconElementRef.nativeElement;
+        this.elementRef.nativeElement.querySelector('.content-layer').firstChild ===
+        this.iconElementRef.nativeElement;
```

Comment:
Alternatively we could consider using viewChild for this. But i'm actually a bit in doubt whether that would improve anything. It just feels like the _angular way™️_ of doing it - which i think is a poor argument for doing it. 

Now i'm just rambling. 

Feel free to do as you wish, this is just my toughts on this line of code 😛 
```
@ViewChild('.content-layer', {read: ElementRef}) contentLayer: ElementRef<HTMLElement>; 
...
this.contentLayer.nativeElement.firstChild 
```

Reply:
I'm not very familiar with `@ViewChild`. I also don't really care how it's implemented as long as I'm testing the correct DOM element 😁, so I'm totally open for suggestions.

Reply:
`@ViewChild` is Angulars mechanism for injecting references to custom components & DOM elements in the hosts template (great 10 minute read about it [here](https://blog.angular-university.io/angular-viewchild/)). 

I think the benefit of utilizing ViewChild is that it's more explicit which elements the controller expects to be present in the template.